client/my-sites/upgrades/checkout/domain-details-form.jsx

```diff
@@ -132,7 +158,7 @@ export default React.createClass( {
 			name,
 			ref: name,
 			additionalClasses: 'checkout-field',
-			value: formState.getFieldValue( this.state.form, name ),
+			value: formState.getFieldValue( this.state.form, name ) || '',
```

Comment:
I prefer to ensure data is in the correct format as early as possible. There's no point in passing `null` to a text field and it'll force the field to have null checks or default values.